adrs/5541831634/README.md

```diff
@@ -0,0 +1,158 @@
+# React project files and directories names conversion
+
+* **State:** Draft
+* **Created:** 2020-12-01
+* **Tags:** react
+
+## Context
+
+Across the ecosystem, we can find all sorts of files and directory naming
+conventions in React projects.
+
+The variety of conventions causes misalignment and shifts focus away from more
+important matters.
+
+Although there are no silver bullets for this topic, committing to a convention
+will reduce cognitive load by triggering pattern recognition, enabling engineers
+to focus their brainpower on their tasks.
+
+## Resolution
+
+* You **must** follow [NodeJS files and directories names conversion](./../3122196229/README.md)
+* You **must** use `<rootDir>/src/@types` to store TypeScript definition files
+  (`*.d.ts`).
+* You **must** use `<rootDir>/src/app.tsx` to be your application root
+  component.
+
+### NextJS
+
+Some resolutions are related to NextJS projects.
+
+* You **must** use `<rootDir>/src/pages` for [NextJS Pages](https://nextjs.org/docs/basic-features/pages)
+  directory.
+* In NextJS project, you **must** re-export a NextJS app from
+  `<rootDir>/src/app.tsx` in `<rootDir>/src/pages/_app.tsx`
+* In NextJS project, you **must** re-export a NextJS Documents from
+  `<rootDir>/src/document.tsx` in `<rootDir>/src/pages/_document.tsx`
+* In NextJS project, you **must** re-export a NextJS Documents from
+  `<rootDir>/src/error.tsx` in `<rootDir>/src/pages/_error.tsx`
+
+### Fractal pattern directories and files
+
+A fractal pattern is a pattern that you could repeat in each nested level
+directory. The pattern repeated in `<rootDir>/src`, `<rootDir>/src/components`
+or `<rootDir>/src/routes/`.
+
+* You **must** use `<fractalDir>/routes` for managing the entry point of a
+  route.
+* You **must** use `<fractalDir>/services` to store anything
+  related to any communication with [external services](#external-services).
+* You **must** use external service brand name as a subdirectory in
+  `<fractalDir>/services/[service name]`.
+* You **must** use `<fractalDir>/components` to store reusable React
```

Comment:
How about for views under components? Should this be a flat structure or is nesting of components under a directory of `/views/${component-name}/Component.${ts|js}.x` advisable?

Reply:
It is mentioned that the Fractal pattern is repeated under `<rootDir>/src/components/`, which means you can keep adding the fractal directory structure under `<rootDir>/src/components`.

Create directories, if you need to scope the content to a particular component or set of components. I updated the docs, let me know if that clarifies your concern or suggest a better wording 😄 

> Side note, make sure to read https://github.com/straw-hat-team/adr/tree/master/adrs/3122196229 since you are using PascalCase in your file name.